### PR TITLE
split packages into their own files

### PR DESCRIPTION
**Pull request by simbabque**

### CODE REVIEW

**Comment by genio** [comment]
Can we merge the pod and pm back together? pod at the end of the file is always good.

**Review by oalders — APPROVED** [review]
Looks great! One thing that comes to mind is that it will be harder to follow the history of changes from files that used to be in `Form.pm`, but it's not an impossible problem to solve.

**Comment by simbabque** [comment]
> Can we merge the pod and pm back together? pod at the end of the file is always good.



The POD is a tricky problem. I think it was an important step towards more modern, maintainable code to split the packages out. But the POD is annoying, because the structure of the documentation is really unusual. I like the way it can be perused on metacpan, you can just search in it, and usually find the right thing. But on the other hand a lot of it is undocumented. There are loads of methods that have no POD.

I considered writing POD for all of the new files, but that would have fragmented the documentation, making it harder to use. I discussed this with @oalders and we agreed that it would be a good middle ground to have it in a separate file. This way it's consistent across all of it. It does mean the POD is not near the methods any more, but at least that's the case for all of them.

Having a POD file for all of the docs seems to be a common paradigm on CPAN.

Can you explain why you'd have it all in Form.pm? Also, do you mean the entire doc at the end, or just the extra bits, with the rest where they belonged?

**Comment by haarg** [comment]
My preference would be to have the entire pod document at the end of lib/HTML/Form.pm.

The split files are missing a newline on the last line.

**Comment by simbabque** [comment]
Sorry, I was away for a few days. So the consensus seems to be to stick it all at the end of the main pm file? That's fine with me. I'll leave another day or two for comments, and then do that, unless someone has another suggestion.

**Comment by oalders** [comment]
> So the consensus seems to be to stick it all at the end of the main pm file?

Right, we discussed this on IRC as well and that was the consensus.

File: Changes [hunk 1]
```diff
@@ -2,6 +2,8 @@ Change history for HTML-Form
 
 {{$NEXT}}
     - Perl::Tidy-ed the entire codebase (GH#42) (Julien Fiegehenn)
+    - Split extra packages inside HTML::Form into their own module files
+      and move the documentation into a POD file (GH#43) (Julien Fiegehenn)
```
Comment (oalders) [inline Changes:6]: I think this needs updating as well.

**Comment by oalders** [comment]
I think once `Changes` has been updated, this is fine to merge.

File: Changes (hunk 1 above)
Comment (simbabque) [reply Changes]: I'm sorry, I thought I had pushed this, but I had not.

**Review by haarg — APPROVED** [review]

**Pull request merged**